registry/remote/repository_test.go

```diff
@@ -2146,3 +2146,63 @@ func Test_ManifestStore_FetchReference(t *testing.T) {
 		t.Errorf("Manifests.FetchReference() = %v, want %v", got, manifest)
 	}
 }
+
+func Test_ZeroSizedBlob_Fetch(t *testing.T) {
+	blob := []byte("")
+	blobDesc := ocispec.Descriptor{
+		MediaType: "test",
+		Digest:    digest.FromBytes(blob),
+		Size:      int64(len(blob)),
+	}
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			t.Errorf("unexpected access: %s %s", r.Method, r.URL)
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		rangeHeader := r.Header.Get("Range")
+		if rangeHeader != "" {
+			rangeLimit, err := strconv.Atoi(rangeHeader[8:])
+			if err == nil && rangeLimit < 0 {
+				t.Errorf("invalid range limit: 0-%d", rangeLimit)
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+		}
```

Comment:
Since we fail as long as the range header presents, we can simplify the mock as

```go
if rangeHeader != "" {
    t.Errorf("unexpected range header: %s", rangeHeader )
    w.WriteHeader(http.StatusBadRequest)
    return
}
```

Reply:
It's not the case that we fail as long as the range header presents. We only fail when the range header presents, and it's of the form of "bytes=0-(a negative number)". When the header is of the form "bytes=0-" (acceptable according to line 626) and "bytes=0-(a positive number)", we don't fail.

Reply:
First of all, the range header check should happen only if the blob exists. In other words, this block of code should in the switch case between lines 2173 and 2174.
Next, as long as the blob to be fetched has zero length, the server returns an error if the range header presents. The error is either 400 or 416. Nonetheless, the unit test fails anyway for 400 or 416.

Reply:
Moved the range header check inside the switch. Differentiated the cases of 400 and 416. The server now returns 400 when the blob has 0 length and a range header is present.